spec/controllers/spree/admin/digitals_controller_spec.rb

```diff
@@ -56,6 +56,18 @@
         end.to change(Spree::Digital, :count).by(1)
       end
     end
+
+    context 'when no asset is attached' do
+      it 'redirects to the index page' do
+        expect do
+          post :create, params: {
+            product_id: product.slug,
+            digital: { variant_id: product.master.id }
+          } # fail validation by not passing attachment
+          expect(response).to redirect_to(spree.admin_product_digitals_path(product))
```

Comment:
@MatthewKennedy shouldn't this fallback to the digital asset form with validation errors?